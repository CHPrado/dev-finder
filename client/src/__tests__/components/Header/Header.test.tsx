import React from "react";
import { BrowserRouter } from "react-router-dom";

import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { Header } from "../../../components";

describe("rendering component", () => {
  it("renders Header component without crashing", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    ).render();
    const title = wrapper.find("h4").text();

    expect(title).toEqual("lookingforDev");
  });
});

describe("snapshots", () => {
  it("Header snapshots", () => {
    const tree = shallow(<Header />);

    expect(toJson(tree)).toMatchSnapshot();
  });
});
