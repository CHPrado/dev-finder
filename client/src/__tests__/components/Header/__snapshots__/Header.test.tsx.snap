// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Header snapshots 1`] = `
<Styled(MuiBox)
  className="makeStyles-root-1"
>
  <Link
    className="makeStyles-link-4"
    to="/"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-3"
      variant="h4"
    >
      <img
        alt="avatar"
        className="makeStyles-logo-2"
        src="githubLogo.png"
      />
      lookingforDev
      <GitBranchIcon
        className="octicon octicon-git-branch"
        size={40}
        verticalAlign="text-bottom"
      />
    </WithStyles(ForwardRef(Typography))>
  </Link>
</Styled(MuiBox)>
`;
